java/node/src/main/java/com/codeheadsystems/dstore/node/dao/TenantDao.java

```diff
@@ -16,107 +16,73 @@
+
   }
 
   /**
-   * Creates the tenant in the database. If it already exists, does nothing but returns the existing tenant.
+   * Creates the tenant in the database. If it already exists, it does nothing but returns the existing tenant.
    *
    * @param tenant to create.
    * @return The tenant... either the one that was created or the existing one.
    */
   public Tenant create(final Tenant tenant) {
     LOGGER.trace("create({})", tenant);
-    sqlEngine.executePreparedInternal("insert into NODE_TENANT (RID_TENANT,UUID,KEY,NONCE) values (?,?,?,?)", (ps) -> {
-      try {
-        ps.setString(1, tenant.id());
-        ps.setString(2, tenant.uuid());
-        ps.setString(3, tenant.key());
-        ps.setString(4, tenant.nonce());
-        ps.execute();
-        if (ps.getUpdateCount() != 1) {
-          throw new IllegalArgumentException("Unable to create tenant");
-        }
-      } catch (SQLException e) {
-        throw new IllegalArgumentException("Unable to read a tenant", e);
-      }
-      return null;
-    });
-    return tenant;
-  }
-
-  /**
-   * Reads a tenant from the database result set. Does not advance the cursor.
-   *
-   * @param rs result set to read from. Must be on a row.
-   * @return a tenant.
-   */
-  private Tenant fromResultSet(final ResultSet rs) {
-    try {
-      return ImmutableTenant.builder()
-          .id(rs.getString("RID_TENANT"))
-          .uuid(rs.getString("UUID"))
-          .key(rs.getString("KEY"))
-          .nonce(rs.getString("NONCE"))
-          .build();
-    } catch (SQLException e) {
-      throw new IllegalArgumentException("Unable to read tenant", e);
+    Integer updateCount = jdbi().withHandle(handle ->
+        handle.createUpdate("insert into NODE_TENANT (RID_TENANT,UUID,KEY,NONCE) values (:ridTenant, :uuid, :key, :nonce)")
+            .bindPojo(tenant)
+            .execute()
+    );
+    if (updateCount != 1) {
+      throw new IllegalArgumentException("Unable to create tenant");
     }
+    return tenant;
   }
 
   /**
-   * Reads from the current database, if the tenant exists.
+   * Reads from the current database if the tenant exists.
    *
    * @param tenantId tenant to read.
    * @return optional tenant if it exists.
    */
   public Optional<Tenant> read(final String tenantId) {
     LOGGER.trace("read({})", tenantId);
-    return sqlEngine.executePreparedInternal("select * from NODE_TENANT where RID_TENANT = ?", (ps) -> {
-      try {
-        ps.setString(1, tenantId);
-        try (final ResultSet rs = ps.executeQuery()) {
-          if (rs.next()) {
-            return Optional.of(fromResultSet(rs));
-          } else {
-            return Optional.empty();
-          }
-        }
-      } catch (SQLException e) {
-        throw new IllegalArgumentException("Unable to read a tenant", e);
-      }
-    });
+    return jdbi().withHandle(handle ->
+        handle.createQuery("select * from NODE_TENANT where RID_TENANT = :tenantId")
```

Comment:
We probably can just rename RID_TENANT to TENANT_ID across the board. (All liquibase) Maybe if this works elsewhere.

Reply:
WIll make the change in a separate commit.